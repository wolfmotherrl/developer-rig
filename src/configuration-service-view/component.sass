.configuration-service-view
  display: grid
  grid-template-columns: 1fr 8pt 1fr
  grid-gap: 8pt
  margin-top: 8pt

  &__button
    background-color: #6441A4
    border: 0
    color: white
    height: 15pt
    margin-left: 8pt
    min-width: 8em

    &--first
      margin-left: 0

  &__title
    font-size: larger
    font-weight: bold

  &-property
    display: block
    margin: 1em

    &__input
      width: 70%

      &--error
        border-color: red

    &__name
      margin-bottom: 5px
      margin-right: 11px

    &__value
      color: grey
      user-select: text

  &__vertical-bar
    background-color: #DAD8DE
    width: 1pt;
    height: 100%

  &__item
    margin-bottom: 8pt
    margin-left: -17pt

  &__item-text
    margin: 3pt

    &--title
      font-weight: bold
