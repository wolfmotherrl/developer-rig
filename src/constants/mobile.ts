export const MobileOrientation: { [key: string]: string } = {
  Portrait: 'Portrait',
  Landscape: 'Landscape',
}

export const DefaultMobileOrientation = MobileOrientation.Portrait;

export const MobileSizes: { [key: string]: { width: number; height: number; } } = Object.freeze({
  'iPhone X/XS Max (414x896)': {
    width: 414,
    height: 896
  },
  'iPhone X/XS (375x822)': {
    width: 375,
    height: 822
  },
  'iPhone 7/8 (375x667)': {
    width: 375,
    height: 667
  },
  'iPhone 7/8 Plus (414x736)': {
    width: 414,
    height: 736
  },
  'Google Pixel 2 (411x731)': {
    width: 411,
    height: 731
  },
  'Samsung Galaxy S8/S9 (360x740)': {
    width: 360,
    height: 740
  },
  'One Plus 3 (480x853)': {
    width: 480,
    height: 853
  },
});

export const DefaultMobileSize = 'iPhone X/XS (375x822)';

