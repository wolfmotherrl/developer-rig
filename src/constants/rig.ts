export const RigRole = 'rig';
export const DeveloperRigUserId = '265737932';

export enum LocalStorageKeys {
  CurrentProjectIndex = 'currentProjectIndex',
  ExtensionViews = 'extensionViews',
  Projects = 'projects',
  RigLogin = 'rigLogin',
}
