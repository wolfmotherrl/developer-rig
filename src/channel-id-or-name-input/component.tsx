import * as React from 'react';
import { DeveloperRigUserId } from '../constants/rig';

interface Props {
  isUser?: boolean;
  name: string;
  value: string;
  labelClassName?: string;
  textClassName?: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
}

interface State {
  checked: boolean;
  lastChannelId: string;
}


export class ChannelIdOrNameInput extends React.Component<Props, State> {
  state: State = {
    checked: this.props.value === DeveloperRigUserId,
    lastChannelId: '',
  }

  public render() {
    return (
      <>
        <label className={this.props.labelClassName}>
          <span className={this.props.textClassName}>
            {this.props.isUser ? 'User' : 'Channel'} ID or Name
          </span>
          <input type="text" name={this.props.name} value={this.props.value} onChange={this.props.onChange} disabled={this.state.checked} />
        </label>
        <label className={this.props.labelClassName}>
          <input type="checkbox" name="wantsDeveloperRigUserId" onChange={this.onChange} checked={this.state.checked} />
          Use Developer Rig user
        </label>
      </>
    );
  }

  private onChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { checked } = event.currentTarget;
    if (checked) {
      this.setState({ lastChannelId: this.props.value });
      this.props.onChange({ currentTarget: { name: this.props.name, value: DeveloperRigUserId } } as React.FormEvent<HTMLInputElement>);
    } else {
      this.props.onChange({ currentTarget: { name: this.props.name, value: this.state.lastChannelId } } as React.FormEvent<HTMLInputElement>);
    }
    this.setState({ checked });
  }
}
