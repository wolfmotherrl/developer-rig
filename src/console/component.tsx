import * as React from 'react';
import './component.sass';

interface ConsoleEntry {
  frame: string;
  log: string;
}

interface State {
  logHistory: ConsoleEntry[];
}

export class Console extends React.Component<{}, State> {
  private console: React.RefObject<HTMLDivElement>;

  public state: State = {
    logHistory: window.rig.history || [],
  }

  constructor(props: {}) {
    super(props);
    this.console = React.createRef();
  }

  private updateConsole() {
    setTimeout(() => {
      const console = this.console.current;
      if (console) {
        const { offsetHeight, scrollHeight } = console;
        console.scrollTop = Math.max(0, scrollHeight - offsetHeight);
      }
    });
    this.setState({
      logHistory: window.rig.history
    });
  }

  public componentDidMount() {
    window.rig.update = () => {
      this.updateConsole();
    }

    this.updateConsole();
  }

  public render() {
    const logs = this.state.logHistory.map((entry, index) => (
      <div key={index}>{entry.frame} $ {entry.log}</div>
    ));

    return (
      <div className="console" ref={this.console}>{logs}</div>
    );
  }
}
