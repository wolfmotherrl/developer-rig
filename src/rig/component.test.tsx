import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { createViewsForTest, createExtensionManifestForTest } from '../tests/constants/extension';
import { mockFetchForUserInfo } from '../tests/mocks';
import { NavItem } from '../constants/nav-items';
import { RigComponent } from './component';
import { ExtensionAnchors } from '../constants/extension-types';
import { ViewerTypes } from '../constants/viewer-types';
import { RigExtensionView, RigProject } from '../core/models/rig';
import { ExtensionViewDialogState } from '../extension-view-dialog';
import { ExtensionAnchor, ExtensionViewType } from '../constants/extension-coordinator';
import { DefaultMobileSize } from '../constants/mobile';

let globalAny = global as any;

localStorage.setItem('projects', '[{"manifest":{}},{"manifest":{}}]');

const setupShallow = setupShallowTest(RigComponent, () => ({
  session: { displayName: 'test', login: 'test', id: 'test', profileImageUrl: 'test.png', authToken: 'test' },
  saveManifest: jest.fn(),
  userLogin: jest.fn(),
}));

describe('<RigComponent />', () => {
  function setUpProjectForTest(type: ExtensionAnchor) {
    const extensionViews = createViewsForTest(1, ExtensionAnchors[type], ViewerTypes.LoggedOut);
    localStorage.setItem('projects', JSON.stringify([{ extensionViews, manifest: {} }]));
    localStorage.setItem('currentProjectIndex', '0');
  }

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      try {
        expect(wrapper).toMatchSnapshot();
        resolve();
      } catch (ex) {
        reject(ex.message);
      }
    });
  });

  it('renders extension view correctly', async () => {
    setUpProjectForTest(ExtensionAnchor.Panel);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          expect(wrapper).toMatchSnapshot();
          expect((wrapper.find('ExtensionViewContainer') as any).props().extensionViews).toHaveLength(1);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('gets extension views from local storage correctly', () => {
    setUpProjectForTest(ExtensionAnchor.Panel);
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          expect(instance.state.currentProject.extensionViews).toEqual(testViews);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('deletes extension view correctly', () => {
    setUpProjectForTest(ExtensionAnchor.Panel);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          expect((wrapper.find('ExtensionViewContainer') as any).props().extensionViews).toHaveLength(1);
          instance.deleteExtensionView('1');
          wrapper.update();
          expect((wrapper.find('ExtensionViewContainer') as any).props().extensionViews).toHaveLength(0);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('toggles state when edit dialog is opened/closed', () => {
    setUpProjectForTest(ExtensionAnchor.Component);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          instance.openEditViewHandler('1');
          expect(instance.state.viewForEdit).toBeTruthy();
          instance.closeEditViewHandler();
          expect(instance.state.viewForEdit).toBe(null);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('edit changes the view and sets them correctly', () => {
    setUpProjectForTest(ExtensionAnchor.Component);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          const views = instance.state.currentProject.extensionViews;
          const editedView = views.filter((element: RigExtensionView) => element.id === '1')[0];
          instance.openEditViewHandler('1');
          expect(instance.state.viewForEdit).toBe(editedView);
          instance.editViewHandler(editedView, { x: 25, y: 25 });
          expect(editedView.x).toEqual(25);
          expect(editedView.y).toEqual(25);
          expect(instance.state.viewForEdit).toBe(null);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('correctly toggles state when create extension view is opened/closed', () => {
    setUpProjectForTest(ExtensionAnchor.Panel);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          instance.state.currentProject = { manifest: createExtensionManifestForTest() } as RigProject;
          instance.openExtensionViewHandler();
          expect(instance.state.showingExtensionsView).toBe(true);
          instance.closeExtensionViewDialog();
          expect(instance.state.showingExtensionsView).toBe(false);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('correctly sets state when viewHandler is invoked', () => {
    setUpProjectForTest(ExtensionAnchor.Panel);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          wrapper.setState({ currentProject: { manifest: createExtensionManifestForTest() } as RigProject });
          const instance = wrapper.instance() as RigComponent;
          instance.viewerHandler(NavItem.ExtensionViews);
          expect(instance.state.selectedView).toBe(NavItem.ExtensionViews);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('gets the correct views when getExtensionViews invoked', () => {
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    setUpProjectForTest(ExtensionAnchor.Panel);
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          expect(instance.state.currentProject.extensionViews).toEqual(testViews);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('returns correct data for mobile ', () => {
    const { wrapper } = setupShallow();
    const testDialogState = {
      width: 0,
      height: 0,
      mobileFrameSize: DefaultMobileSize,
      extensionViewType: ExtensionViewType.Mobile
    } as ExtensionViewDialogState;
    const expectedMobileFrameSize = {
      width: 375,
      height: 822,
    };
    const instance = wrapper.instance() as RigComponent;

    let frameSize = instance.getFrameSizeFromDialog(testDialogState);
    expect(frameSize).toEqual(expectedMobileFrameSize);
  });

  it('returns correct data for other types', () => {
    const { wrapper } = setupShallow();
    const overlayTestDialogState = {
      width: 0,
      height: 0,
      frameSize: '640x480',
      extensionViewType: ExtensionViewType.Overlay
    } as ExtensionViewDialogState;
    const expectedOverlayFrameSize = {
      width: 640,
      height: 480,
    };
    const instance = wrapper.instance() as RigComponent;
    const frameSize = instance.getFrameSizeFromDialog(overlayTestDialogState);
    expect(frameSize).toEqual(expectedOverlayFrameSize);
  });

  it('returns correct data for custom size', () => {
    const { wrapper } = setupShallow();
    const overlayTestDialogState = {
      width: 100,
      height: 100,
      frameSize: 'Custom',
      extensionViewType: ExtensionViewType.Overlay
    } as ExtensionViewDialogState;
    const expectedOverlayFrameSize = {
      width: 100,
      height: 100,
    };
    const instance = wrapper.instance() as RigComponent;
    const frameSize = instance.getFrameSizeFromDialog(overlayTestDialogState);
    expect(frameSize).toEqual(expectedOverlayFrameSize);
  });

  it('correctly fetches user info if login not in localStorage', () => {
    globalAny.fetch = jest.fn().mockImplementation(mockFetchForUserInfo);
    globalAny.window.location.hash = 'access_token=test&';

    setupShallow();
    expect(globalAny.fetch).toHaveBeenCalled();
  });

  globalAny.confirm = jest.fn().mockImplementation(() => true);
  globalAny.fetch = jest.fn().mockImplementation(() => Promise.resolve({
    status: 200,
    json: () => Promise.resolve({}),
  }));

  it('creates project', async () => {
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(async () => {
        try {
          wrapper.update();
          const instance = wrapper.instance() as RigComponent;
          const extensionViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
          await instance.createProject({ extensionViews } as RigProject);
          expect(globalAny.fetch).toHaveBeenCalled();
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('selects project', async () => {
    setUpProjectForTest(ExtensionAnchor.Panel);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;
    await instance.selectProject(1);
    expect(globalAny.fetch).toHaveBeenCalled();
  });

  it('deletes project', async () => {
    const { wrapper } = setupShallow();
    return new Promise((resolve, reject) => {
      setTimeout(async () => {
        try {
          wrapper.update();
          const currentProject = { manifest: createExtensionManifestForTest() } as RigProject;
          wrapper.setState({ currentProject });
          const instance = wrapper.instance() as RigComponent;
          await instance.deleteProject();
          expect(globalAny.confirm).toHaveBeenCalled();
          expect(instance.state.currentProject).not.toBe(currentProject);
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('correctly displays an error if user fetch fails', done => {
    const errorMessage = 'test';
    globalAny.fetch = jest.fn().mockImplementation(() => Promise.reject(new Error(errorMessage)));
    globalAny.window.location.hash = 'access_token=test&';
    const { wrapper } = setupShallow();
    setTimeout(() => {
      wrapper.update();
      expect(globalAny.fetch).toHaveBeenCalled();
      expect(wrapper.instance().state.error).toEqual(errorMessage);
      done();
    });
  });
});
