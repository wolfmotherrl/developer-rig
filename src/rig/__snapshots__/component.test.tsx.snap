// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigComponent /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigComponent
    saveManifest={[Function]}
    session={
        Object {
            "authToken": "test",
            "displayName": "test",
            "id": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
    userLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(RigNavComponent)
          createNewProject={[Function]}
          currentProjectIndex={-1}
          deleteProject={[Function]}
          error={undefined}
          manifest={null}
          projects={Array []}
          selectProject={[Function]}
          selectedView={2}
          viewerHandler={[Function]}
/>,
        <Unknown>
          <CreateProjectDialog
                    closeHandler={[Function]}
                    mustSave={true}
                    saveHandler={[Function]}
                    userId={undefined}
          />
</Unknown>,
      ],
      "className": "rig-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createNewProject": [Function],
          "currentProjectIndex": -1,
          "deleteProject": [Function],
          "error": undefined,
          "manifest": null,
          "projects": Array [],
          "selectProject": [Function],
          "selectedView": 2,
          "viewerHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            false,
            undefined,
            undefined,
            undefined,
            <CreateProjectDialog
              closeHandler={[Function]}
              mustSave={true}
              saveHandler={[Function]}
              userId={undefined}
/>,
            false,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          undefined,
          undefined,
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "closeHandler": [Function],
              "mustSave": true,
              "saveHandler": [Function],
              "userId": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          undefined,
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(RigNavComponent)
            createNewProject={[Function]}
            currentProjectIndex={-1}
            deleteProject={[Function]}
            error={undefined}
            manifest={null}
            projects={Array []}
            selectProject={[Function]}
            selectedView={2}
            viewerHandler={[Function]}
/>,
          <Unknown>
            <CreateProjectDialog
                        closeHandler={[Function]}
                        mustSave={true}
                        saveHandler={[Function]}
                        userId={undefined}
            />
</Unknown>,
        ],
        "className": "rig-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createNewProject": [Function],
            "currentProjectIndex": -1,
            "deleteProject": [Function],
            "error": undefined,
            "manifest": null,
            "projects": Array [],
            "selectProject": [Function],
            "selectedView": 2,
            "viewerHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              false,
              undefined,
              undefined,
              undefined,
              <CreateProjectDialog
                closeHandler={[Function]}
                mustSave={true}
                saveHandler={[Function]}
                userId={undefined}
/>,
              false,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            undefined,
            undefined,
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "closeHandler": [Function],
                "mustSave": true,
                "saveHandler": [Function],
                "userId": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            undefined,
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RigComponent /> renders extension view correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigComponent
    saveManifest={[Function]}
    session={
        Object {
            "authToken": "test",
            "displayName": "test",
            "id": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
    userLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(RigNavComponent)
          createNewProject={[Function]}
          currentProjectIndex={0}
          deleteProject={[Function]}
          error={undefined}
          manifest={Object {}}
          projects={
                    Array [
                              Object {
                                "extensionViews": Array [
                                  Object {
                                    "channelId": "twitch",
                                    "features": Object {
                                      "isChatEnabled": true,
                                    },
                                    "id": "1",
                                    "isPopout": false,
                                    "linked": false,
                                    "linkedUserId": "",
                                    "mode": "viewer",
                                    "opaqueId": "ARIGopaqueId",
                                    "orientation": "Portrait",
                                    "role": "Logged-Out Viewer",
                                    "type": "Panel",
                                    "x": 0,
                                    "y": 0,
                                  },
                                ],
                                "manifest": Object {},
                              },
                            ]
          }
          selectProject={[Function]}
          selectedView={0}
          viewerHandler={[Function]}
/>,
        <Unknown>
          <ExtensionViewContainer
                    configurations={undefined}
                    deleteExtensionViewHandler={[Function]}
                    extensionViews={
                              Array [
                                        Object {
                                          "channelId": "twitch",
                                          "features": Object {
                                            "isChatEnabled": true,
                                          },
                                          "id": "1",
                                          "isPopout": false,
                                          "linked": false,
                                          "linkedUserId": "",
                                          "mode": "viewer",
                                          "opaqueId": "ARIGopaqueId",
                                          "orientation": "Portrait",
                                          "role": "Logged-Out Viewer",
                                          "type": "Panel",
                                          "x": 0,
                                          "y": 0,
                                        },
                                      ]
                    }
                    isDisplayed={true}
                    isLocal={undefined}
                    manifest={Object {}}
                    openEditViewHandler={[Function]}
                    openExtensionViewHandler={[Function]}
                    secret={undefined}
          />
</Unknown>,
      ],
      "className": "rig-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createNewProject": [Function],
          "currentProjectIndex": 0,
          "deleteProject": [Function],
          "error": undefined,
          "manifest": Object {},
          "projects": Array [
            Object {
              "extensionViews": Array [
                Object {
                  "channelId": "twitch",
                  "features": Object {
                    "isChatEnabled": true,
                  },
                  "id": "1",
                  "isPopout": false,
                  "linked": false,
                  "linkedUserId": "",
                  "mode": "viewer",
                  "opaqueId": "ARIGopaqueId",
                  "orientation": "Portrait",
                  "role": "Logged-Out Viewer",
                  "type": "Panel",
                  "x": 0,
                  "y": 0,
                },
              ],
              "manifest": Object {},
            },
          ],
          "selectProject": [Function],
          "selectedView": 0,
          "viewerHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            false,
            false,
            undefined,
            <ExtensionViewContainer
              configurations={undefined}
              deleteExtensionViewHandler={[Function]}
              extensionViews={
                            Array [
                                          Object {
                                            "channelId": "twitch",
                                            "features": Object {
                                              "isChatEnabled": true,
                                            },
                                            "id": "1",
                                            "isPopout": false,
                                            "linked": false,
                                            "linkedUserId": "",
                                            "mode": "viewer",
                                            "opaqueId": "ARIGopaqueId",
                                            "orientation": "Portrait",
                                            "role": "Logged-Out Viewer",
                                            "type": "Panel",
                                            "x": 0,
                                            "y": 0,
                                          },
                                        ]
              }
              isDisplayed={true}
              isLocal={undefined}
              manifest={Object {}}
              openEditViewHandler={[Function]}
              openExtensionViewHandler={[Function]}
              secret={undefined}
/>,
            false,
            false,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          undefined,
          Object {
            "instance": null,
            "key": "ExtensionViewContainer0",
            "nodeType": "class",
            "props": Object {
              "configurations": undefined,
              "deleteExtensionViewHandler": [Function],
              "extensionViews": Array [
                Object {
                  "channelId": "twitch",
                  "features": Object {
                    "isChatEnabled": true,
                  },
                  "id": "1",
                  "isPopout": false,
                  "linked": false,
                  "linkedUserId": "",
                  "mode": "viewer",
                  "opaqueId": "ARIGopaqueId",
                  "orientation": "Portrait",
                  "role": "Logged-Out Viewer",
                  "type": "Panel",
                  "x": 0,
                  "y": 0,
                },
              ],
              "isDisplayed": true,
              "isLocal": undefined,
              "manifest": Object {},
              "openEditViewHandler": [Function],
              "openExtensionViewHandler": [Function],
              "secret": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          undefined,
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(RigNavComponent)
            createNewProject={[Function]}
            currentProjectIndex={0}
            deleteProject={[Function]}
            error={undefined}
            manifest={Object {}}
            projects={
                        Array [
                                    Object {
                                      "extensionViews": Array [
                                        Object {
                                          "channelId": "twitch",
                                          "features": Object {
                                            "isChatEnabled": true,
                                          },
                                          "id": "1",
                                          "isPopout": false,
                                          "linked": false,
                                          "linkedUserId": "",
                                          "mode": "viewer",
                                          "opaqueId": "ARIGopaqueId",
                                          "orientation": "Portrait",
                                          "role": "Logged-Out Viewer",
                                          "type": "Panel",
                                          "x": 0,
                                          "y": 0,
                                        },
                                      ],
                                      "manifest": Object {},
                                    },
                                  ]
            }
            selectProject={[Function]}
            selectedView={0}
            viewerHandler={[Function]}
/>,
          <Unknown>
            <ExtensionViewContainer
                        configurations={undefined}
                        deleteExtensionViewHandler={[Function]}
                        extensionViews={
                                    Array [
                                                Object {
                                                  "channelId": "twitch",
                                                  "features": Object {
                                                    "isChatEnabled": true,
                                                  },
                                                  "id": "1",
                                                  "isPopout": false,
                                                  "linked": false,
                                                  "linkedUserId": "",
                                                  "mode": "viewer",
                                                  "opaqueId": "ARIGopaqueId",
                                                  "orientation": "Portrait",
                                                  "role": "Logged-Out Viewer",
                                                  "type": "Panel",
                                                  "x": 0,
                                                  "y": 0,
                                                },
                                              ]
                        }
                        isDisplayed={true}
                        isLocal={undefined}
                        manifest={Object {}}
                        openEditViewHandler={[Function]}
                        openExtensionViewHandler={[Function]}
                        secret={undefined}
            />
</Unknown>,
        ],
        "className": "rig-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createNewProject": [Function],
            "currentProjectIndex": 0,
            "deleteProject": [Function],
            "error": undefined,
            "manifest": Object {},
            "projects": Array [
              Object {
                "extensionViews": Array [
                  Object {
                    "channelId": "twitch",
                    "features": Object {
                      "isChatEnabled": true,
                    },
                    "id": "1",
                    "isPopout": false,
                    "linked": false,
                    "linkedUserId": "",
                    "mode": "viewer",
                    "opaqueId": "ARIGopaqueId",
                    "orientation": "Portrait",
                    "role": "Logged-Out Viewer",
                    "type": "Panel",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "manifest": Object {},
              },
            ],
            "selectProject": [Function],
            "selectedView": 0,
            "viewerHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              false,
              false,
              undefined,
              <ExtensionViewContainer
                configurations={undefined}
                deleteExtensionViewHandler={[Function]}
                extensionViews={
                                Array [
                                                Object {
                                                  "channelId": "twitch",
                                                  "features": Object {
                                                    "isChatEnabled": true,
                                                  },
                                                  "id": "1",
                                                  "isPopout": false,
                                                  "linked": false,
                                                  "linkedUserId": "",
                                                  "mode": "viewer",
                                                  "opaqueId": "ARIGopaqueId",
                                                  "orientation": "Portrait",
                                                  "role": "Logged-Out Viewer",
                                                  "type": "Panel",
                                                  "x": 0,
                                                  "y": 0,
                                                },
                                              ]
                }
                isDisplayed={true}
                isLocal={undefined}
                manifest={Object {}}
                openEditViewHandler={[Function]}
                openExtensionViewHandler={[Function]}
                secret={undefined}
/>,
              false,
              false,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            undefined,
            Object {
              "instance": null,
              "key": "ExtensionViewContainer0",
              "nodeType": "class",
              "props": Object {
                "configurations": undefined,
                "deleteExtensionViewHandler": [Function],
                "extensionViews": Array [
                  Object {
                    "channelId": "twitch",
                    "features": Object {
                      "isChatEnabled": true,
                    },
                    "id": "1",
                    "isPopout": false,
                    "linked": false,
                    "linkedUserId": "",
                    "mode": "viewer",
                    "opaqueId": "ARIGopaqueId",
                    "orientation": "Portrait",
                    "role": "Logged-Out Viewer",
                    "type": "Panel",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "isDisplayed": true,
                "isLocal": undefined,
                "manifest": Object {},
                "openEditViewHandler": [Function],
                "openExtensionViewHandler": [Function],
                "secret": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
            undefined,
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
